src/rustup.ts

```diff
@@ -213,21 +218,59 @@ export function parseActiveToolchain(rustupOutput: string): string {
  * Returns active (including local overrides) toolchain, as specified by rustup.
  * May throw if rustup at specified path can't be executed.
  */
-export function getActiveChannel(rustupPath: string, wsPath: string): string {
+export function getActiveChannel(wsPath: string, config: RustupConfig): string {
     // rustup info might differ depending on where it's executed
     // (e.g. when a toolchain is locally overriden), so executing it
     // under our current workspace root should give us close enough result
 
     let activeChannel;
     try {
         // `rustup show active-toolchain` is available since rustup 1.12.0
-        activeChannel = child_process.execSync(`${rustupPath} show active-toolchain`, { cwd: wsPath }).toString().trim();
+        activeChannel = runRustupSync(config.path, ['show', 'active-toolchain'], { cwd: wsPath }, config.useWSL).toString().trim();
     } catch (e) {
         // Possibly an old rustup version, so try rustup show
-        const showOutput = child_process.execSync(`${rustupPath} show`, { cwd: wsPath }).toString();
+        const showOutput = runRustupSync(config.path, ['show'], { cwd: wsPath }, config.useWSL).toString();
         activeChannel = parseActiveToolchain(showOutput);
     }
 
     console.info(`Detected active channel: ${activeChannel} (since 'rust-client.channel' is unspecified)`);
     return activeChannel;
 }
+
+export async function runRustup(rustup: string, args: string[], options: child_process.ExecFileOptions, useWSL?: boolean): Promise<ExecChildProcessResult> {
+    let command: string = rustup;
+
+    if (useWSL == true) {
+        ({ command: command, args: args } = modifyParametersForWSL(command, args));
+    }
+
+    return execFile(command, args, options);
+}
+
+export function runRustupSync(rustup: string, args: string[], options: child_process.ExecFileOptions, useWSL?: boolean): Buffer {
+    let command: string = rustup;
+
+    if (useWSL == true) {
+        ({ command: command, args: args } = modifyParametersForWSL(command, args));
+    }
+
+    return child_process.execFileSync(command, args, { ...options });
+}
+
+export function runRustupProcess(rustup: string, args: string[], options: child_process.ExecFileOptions, useWSL?: boolean): child_process.ChildProcess {
```

Comment:
Same as above

```suggestion
export function runProcess(command: string, args: string[], options: child_process.ExecFileOptions, useWSL?: boolean): child_process.ChildProcess {
```

Reply:
`spawnProcess` seems better. Internally it calls `spawn()` from `Node.js` so some with that background will have an idea what that function is doing just by a name

Reply:
Sounds good!